scripts/generate_dump

```diff
@@ -537,17 +732,77 @@ collect_mellanox() {
 #  None
 ###############################################################################
 collect_broadcom() {
-    save_cmd "bcmcmd -t5 version" "broadcom.version"
-    save_cmd "bcmcmd -t5 soc" "broadcom.soc"
-    save_cmd "bcmcmd -t5 ps" "broadcom.ps"
+    local platform=$(show platform summary --json | python -c 'import sys, json; \
+        print(json.load(sys.stdin)["platform"])')
+    local hwsku=$(show platform summary --json | python -c 'import sys, json; \
+        print(json.load(sys.stdin)["hwsku"])')
+
+    # save SAI configuration files (config.bcm, port_config.ini, sai.profile)
+    if [ -d /usr/share/sonic/device/${platform}/${hwsku} ]; then
+        # copy all the files in the HWSKU directory
+        pushd /usr/share/sonic/device/${platform}/${hwsku} > /dev/null
+        for file in $(find . -maxdepth 2 -type f); do
+            save_file ${file} sai false
+        done
+        popd > /dev/null
+
+        if [[ ("$NUM_ASICS" > 1) ]]; then
+            for (( i=0; i<$NUM_ASICS; i++ ))
+            do
+                # config.bcm - copy the one with chip common properties merged
+                for file in $(find /var/run/docker-syncd$i -type f -name "*.bcm"); do
+                    save_file ${file} sai$i false
+                done
+                # sai.profile - copy the final sai.profile generated in docker-syncd
+                if [ -f /var/run/docker-syncd$i/sai.profile ]; then
+                    save_file /var/run/docker-syncd$i/sai.profile sai$i false
+                fi
+            done
+        else
+            # config.bcm - copy the one with chip common properties merged
+            for file in $(find /var/run/docker-syncd -type f -name "*.bcm"); do
+                save_file ${file} sai false
+            done
+            # sai.profile - copy the final sai.profile generated in docker-syncd
+            if [ -f /var/run/docker-syncd/sai.profile ]; then
+                save_file /var/run/docker-syncd/sai.profile sai false
+            fi
+        fi
+    else
+        echo "'/usr/share/sonic/device/${platform}/${hwsku}' does not exist" > /tmp/error
+        save_file /tmp/error sai false
+    fi
+
     save_cmd "cat /proc/bcm/knet/debug" "broadcom.knet.debug"
     save_cmd "cat /proc/bcm/knet/dma" "broadcom.knet.dma"
-    save_cmd "cat /proc/bcm/knet/dstats" "broadcom.knet.dstats"
     save_cmd "cat /proc/bcm/knet/link" "broadcom.knet.link"
     save_cmd "cat /proc/bcm/knet/rate" "broadcom.knet.rate"
-    save_cmd "cat /proc/bcm/knet/stats" "broadcom.knet.stats"
```

Comment:
not sure why this is removed.

Reply:
@Iguohan
It is not removed. These stats were moved under save_counter_snapshot() function. We need to take two snapshots of the counter stats to get the counter trend.